libs/langchain/langchain/output_parsers/boolean.py

```diff
@@ -19,13 +19,16 @@ def parse(self, text: str) -> bool:
             boolean
 
         """
-        cleaned_text = text.strip()
-        if cleaned_text.upper() not in (self.true_val.upper(), self.false_val.upper()):
+        cleaned_upper_text = text.strip().upper()
+        if self.true_val.upper() in cleaned_upper_text:
```

Comment:
probably worth first checking that both true_val and false_val don't occur in text

Reply:
yup, good point! Added with tests! thanks for checking